src/hooks/useOnEscapePress.ts

```diff
@@ -0,0 +1,22 @@
+
+import {useEffect, useCallback} from 'react'
+
+export type UseOnEscapePressProps = {
+  onEscape: (e: KeyboardEvent) => void
+}
+
+export const useOnEscapePress = ({onEscape}: UseOnEscapePressProps): void => {
+  const handleEscape = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === 'Escape') {
```

Comment:
This handler doesn't check `e.defaultPrevented`, which slightly worries me. Here is the scenario:

- You're in a dialog
- The dialog uses `onEscapeKeyPress` to dismiss
- Inside the dialog, you have a select menu, which shows an overlay on top of the dialog
- The select menu listens for the Escape key press, which hides the overlay
- Since we do not check `e.defaultPrevented` in this hook, a single Escape key press will also dismiss the dialog.

The expected behavior is to dismiss the overlay for the select menu on the first Escape key press, then the dialog on the _second_ key press.

Reply:
This line of thinking also has me wondering if the event listener should be on something other than `document`. I am not sure if calling `preventDefault()` will result in that flag getting set for other event handlers on the same Node. In fact, I don't even know if the order they are called in is guaranteed, and if so, what that order would be. Are most-recently-attached listeners executed first? If so, and if `preventDefault()` actually sets the `defaultPrevented` flag on the rest of the event handlers, then I think we're fine (as long as we actually check for `defaultPrevented`).